sdk/core/azure-core/inc/azure/core/azure_assert.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provide assert macros to use as pre-conditions.
+ *
+ * @remark Asserts are turned ON even if `NDEBUG` is defined.
+ *
+ */
+
+#pragma once
+
+#if defined(NDEBUG)
+#define AZURE_SDK_INTERNAL_IMPLEMENTATION_OVERRIDE_NDEBUG
+#undef NDEBUG
+#endif
+
+#include <cassert>
+#include <stdexcept>
```

Comment:
This include is unused I'd remove it, and failing that include it where `NDEBUG` has the user-provided value instead of here so this header and any others it includes see the user-provided value of `NDEBUG`.